tools/diff-processor/diff/sets_test.go

```diff
@@ -31,3 +33,83 @@ func TestRemoveZeroPadding(t *testing.T) {
 		}
 	}
 }
+
+func TestIsSubsetOf(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		set1     FieldSet
+		set2     FieldSet
+		isSubset bool
+	}{
+		{
+			name:     "empty set is subset of any set",
+			set1:     FieldSet{},
+			set2:     FieldSet{"a": {}},
+			isSubset: true,
+		},
+		{
+			name:     "set is subset of itself",
+			set1:     FieldSet{"a": {}},
+			set2:     FieldSet{"a": {}},
+			isSubset: true,
+		},
+		{
+			name:     "set is not subset of smaller set",
+			set1:     FieldSet{"a": {}, "b": {}},
+			set2:     FieldSet{"a": {}},
+			isSubset: false,
+		},
+		{
+			name:     "set is subset of larger set",
+			set1:     FieldSet{"a": {}},
+			set2:     FieldSet{"a": {}, "b": {}},
+			isSubset: true,
+		},
+	} {
+		if got := tc.set1.IsSubsetOf(tc.set2); got != tc.isSubset {
+			t.Errorf("unexpected result for test case %s: %t (expected %t)", tc.name, got, tc.isSubset)
+		}
+	}
+}
+
+func TestDifference(t *testing.T) {
+	for _, tc := range []struct {
+		name     string
+		set1     FieldSet
+		set2     FieldSet
+		expected FieldSet
+	}{
+		{
+			name:     "empty set difference is empty set",
+			set1:     FieldSet{},
+			set2:     FieldSet{"a": {}},
+			expected: FieldSet{},
+		},
+		{
+			name:     "set difference with itself is empty set",
+			set1:     FieldSet{"a": {}},
+			set2:     FieldSet{"a": {}},
+			expected: FieldSet{},
+		},
+		{
+			name:     "set difference with subset is subset",
```

Comment:
nit: name is inaccurate

```suggestion
			name:     "set difference with subset is diff",
```